From 65c866ec56cdc8d7beea3df5e0fc50d7ad483c93 Mon Sep 17 00:00:00 2001
From: Anuj Mittal <anuj.mittal@intel.com>
Date: Wed, 11 Mar 2020 12:07:12 +0800
Subject: [PATCH] sgx: patch makefile to point to correct ocaml libs

fixes:

    | ocamlc.opt -c -ccopt -fpie -o Ast.cmo Ast.ml
    | + ocamlc.opt -c -ccopt -fpie -o Ast.cmo Ast.ml
    | File "command line", line 1:
    | Error: Unbound module Pervasives
    | Command exited with code 2.
    | Hint: Recursive traversal of subdirectories was not enabled for this build,
    |   as the working directory does not look like an ocamlbuild project (no
    |   '_tags' or 'myocamlbuild.ml' file). If you have modules in subdirectories,
    |   you should add the option "-r" or create an empty '_tags' file.
    |
    |   To enable recursive traversal for some subdirectories only, you can use the
    |   following '_tags' file:
    |
    |       true: -traverse
    |       <dir1> or <dir2>: traverse
    |
    | make: *** [Makefile:48: build] Error 10

Signed-off-by: Anuj Mittal <anuj.mittal@intel.com>
---
 sdk/edger8r/linux/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sdk/edger8r/linux/Makefile b/sdk/edger8r/linux/Makefile
index 4706cacc..9ffde000 100644
--- a/sdk/edger8r/linux/Makefile
+++ b/sdk/edger8r/linux/Makefile
@@ -45,7 +45,7 @@ build:
 ifeq ($(shell test $(OCAML_VERSION) -lt 402 && echo 1), 1)
 	ocamlbuild -lflag -ccopt -lflag "-Wl,-z,now" -no-links -libs str,unix  Edger8r.native
 else
-	ocamlbuild -cflags -ccopt,-fpie -lflags -runtime-variant,_pic,-ccopt,-pie,-ccopt -lflag "-Wl,-z,now"  -no-links -libs str,unix Edger8r.native
+	ocamlbuild -cflags -I,${OCAML_NATIVE_LIB} -lflags -I,${OCAML_NATIVE_LIB} -cflags -ccopt,-fpie -lflags -runtime-variant,_pic,-ccopt,-pie,-ccopt -lflag "-Wl,-z,now"  -no-links -libs str,unix Edger8r.native
 endif
 
 $(BUILD_DIR):
-- 
2.24.1

